src/rgw/rgw_common.cc

```diff
@@ -2196,26 +2196,25 @@ bool match_policy(const std::string& pattern, const std::string& input,
  */
 string lowercase_dash_http_attr(const string& orig, bool bidirection)
 {
-  const char *s = orig.c_str();
-  char buf[orig.size() + 1];
-  buf[orig.size()] = '\0';
-
-  for (size_t i = 0; i < orig.size(); ++i, ++s) {
-    switch (*s) {
-      case '_':
-        buf[i] = '-';
-        break;
-      case '-':
-        if (bidirection)
-          buf[i] = '_';
-        else
-          buf[i] = tolower(*s);
-        break;
-      default:
-        buf[i] = tolower(*s);
-    }
-  }
-  return string(buf);
+  std::string buf;
+  buf.reserve(orig.size());
+
+  std::transform(orig.begin(), orig.cend(),
```

Comment:
Hm. Same thing, but tolower() instead of toupper(). I think this is still an improvement, but it does start my head-scratching!